### ISSUE TITLE
Stopping service tasks with inactive task definitions.

### ISSUE THREAD

**Issue by joaoricardo000** [fabfuel/ecs-deploy#12]
Hi!
I'm having an issue when using the ecs deploy on CircleCI.

After the whole process (create new task def, update service), it freezes on the "Deploying task definition".
```
ecs deploy $AWS_CLUSTER $AWS_SERVICE -t $AWS_TAG_DEV
Updating task definition
Changed image of container 'xxxxx-frontend' to: "1234.dkr.ecr.us-west-2.amazonaws.com/xxxxx/frontend:dev" (was: "1234.dkr.ecr.us-west-2.amazonaws.com/xxxxx/frontend:dev")

Creating new task definition revision
Successfully created revision: 10
Successfully deregistered revision: 9

Updating service
Successfully changed task definition to: xxxxx-frontend:10

Deploying task definition...................................................
```  
  
Checking the AWS Console, I can see that the new task is running, with the new task definition - but the old one is not stopped and the desired status is still 'running'.
If I manually stop the old task, the "Deploying task definition" unfreezes and the process finishes with success.  The desired status is 1 task for this service, so checking your code I can see that the [stop task](http://docs.aws.amazon.com/cli/latest/reference/ecs/stop-task.html) is never called for the old one.  
  
Am I missing something? Is this an error of configuration on ECS?

Thanks for the project!

**Comment by fabfuel** [comment]
Hi @joaoricardo000! 
Does the same thing happen, if you run this locally and not in CircleCI?

Actually this tool never calls `start` or `stop task`. It's just creating a new task definition and updating a service with this new task revision. The actual deployment and task management happens completely on AWS side.

It's strange that it freezes, because this is just a loop, which checks periodically if the service has deployed the new task revision and is running in a steady state.

During your deployment, can you see two entries in the `Deployments` tab on the service's page in the AWS console?

What are the service's deployment options `Minimum healthy percent` and `Maximum percent`?

Do you see any events in the `Events` tab on the service page?

Best
Fabian

**Comment by joaoricardo000** [comment]
Hi @fabfuel!
Thanks for the fast reply.

TL;DR: The deployment process was taking up to 7min to finish - probably due ELB - and the timeout is by default 5min.

We were very impatient thinking something was wrong instead of just sitting and waiting.  
Running this on CircleCI, will take an amount of time that we might not have. Right now I'm thinking in creating a flag to skip the ```wait_for_finish(deployment...```  and just assume AWS will do everything, if I updated the tasks and services. What do you think about this approach?

**Comment by fabfuel** [comment]
I would not skip it. 

IMHO: Whats the point of running CI, if you do not care about the results? :-)
Why can't CircleCI wait for the finish, you can not use the environment anyway until the deployment is done.

As I understood, you are just running one task, right? 
Do you also have only one container instance in the cluster? Then it will take some time.

The deployment is much faster, if you can do a rolling or blue/green deployment, which ECS does by default. If you have more then one container instance, you can achieve that, by setting `Maximum percent` to 200 to run the old and new task revision in parallel for a short amount of time. 

If you only have one container instance, you can achieve that by using dynamic ports, if you are running an application load balancer, instead of the classic ELB.

Best
Fabian

**Comment by fabfuel** [comment]
If you just want to fire and forget the deployment in CIrcleCI, you already can do that by running the process in the background: `ecs deploy clust service &`

**Comment by joaoricardo000** [comment]
We are waiting the ~7min that takes to update the service, not a real problem right now.  The idea to skip only the ```wait for finish``` was to be sure that, at least, task and services were updated successfully. 
Closing this issue because is no longer an issue, just had to use the flag ```--timeout 600```.
